### PR TITLE
[4.0] Subfields - refactor for accessibility etc

### PR DESCRIPTION
**Pull request by brianteeman**
Fix the layout issue as reported with #26602 and numerous other fixes

### summary

- set column widths
- Consistency rename all instances of "sub field" to "sub fields"
- Apply style guide rule for labels
- change the table class and add table id
- replace links with button and remove useless tabindex
- kill the description for the render value field but keep it in the code as a tip (it doesn't need to be repeated on every row)
- Several code style changes
- remove icon-white class from buttons - it was not required and resulted in the icons not being centered

### Notes

- It looks to me from the code that I should be able to have repeatable fieldgroups but I couldnt see how 
- Really not sure why the fields are in a for each loop when there are only ever the same two. Maybe I am missing something

### CODE REVIEW

File: layouts/joomla/form/field/subform/repeatable-table.php [hunk 1]
```diff
@@ -43,11 +43,11 @@
 if (!empty($groupByFieldset))
 {
 	foreach ($tmpl->getFieldsets() as $fieldset) {
-		$table_head .= '<th>' . Text::_($fieldset->label);
+		$table_head .= '<th scope="col" style="width:45%">' . Text::_($fieldset->label);
 
 		if ($fieldset->description)
 		{
-			$table_head .= '<br><small style="font-weight:normal">' . Text::_($fieldset->description) . '</small>';
+			$table_head .= '<span class="fa fa-info-circle" aria-hidden="true" tabindex="0"></span><div role="tooltip" id="tip<?php echo $i; ?>">' . Text::_($field->description) . '</div>';
```
Comment (Quy) [inline layouts/joomla/form/field/subform/repeatable-table.php]: `<?php echo $i; ?>` is inside the string, thus, it is considered text and not executed as PHP code. Same with line 65.

**Comment by Quy** [comment]
> kill the description for the render value field but keep it in the code as a tip (it doesn't need to be repeated on every row)

Still repeated. See under Global Configuration > Users > Email Domain Options



Would it be better to display it once in the table header instead in a tooltip?

**Comment by brianteeman** [comment]
Ah I didnt know it was being used by core anywhere. I will review that later and update this PR. Is core using this anywhere else?

> Would it be better to display it once in the table header instead in a tooltip?

Yes it would

**Comment by brianteeman** [comment]
Now that I see the email options page uses this I can see even more issues with the original code - how the heck was this tested and merged

File: layouts/joomla/form/field/subform/repeatable-table.php (hunk 1 above)
Comment (brianteeman) [reply layouts/joomla/form/field/subform/repeatable-table.php]: fixed

**Comment by Quy** [comment]
Missing language string under Global Configuration > Users > Email Domain Options.
```			
<caption id="captionTable" class="sr-only">
				PLG_FIELDS_SUBFIELDS_TABLE_CAPTION			</caption>
```

> Is core using this anywhere else?

Under Plugins > System - Redirect, but layout is slightly different. I don't know if there are others.

**Comment by AndySDH** [comment]
> Now that I see the email options page uses this I can see even more issues with the original code - how the heck was this tested and merged

See this: https://github.com/joomla/joomla-cms/pull/24711

This was required to be merged by the PR creator for funcionallity wise as over a year of work went into it.

Layout/appearance wise it does indeed still need work.

**Comment by brianteeman** [comment]
> Layout/appearance wise it does indeed still need work.

Not just that as seen from the long list :)

**Comment by jduerscheid** [comment]
@brianteeman: Please add some testinstructions<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26614">issues.joomla.org/tracker/joomla-cms/26614</a>.</sub>

**Comment by uglyeoin** [comment]
I have tested this item :white_check_mark: successfully on b8403d70d7b81bcd5d6611cee675268e134c06ad<br /><br />All seems to work well.  <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26614">issues.joomla.org/tracker/joomla-cms/26614</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b8403d70d7b81bcd5d6611cee675268e134c06ad<br /><br />The repeating description can be addressed in a separate PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26614">issues.joomla.org/tracker/joomla-cms/26614</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26614">issues.joomla.org/tracker/joomla-cms/26614</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

@Quy can you open an issue for the repeated description please

**Comment by AndySDH** [comment]
The repeated description in the Email Domain Options page is still an issue in J4 Beta

**Comment by wilsonge** [comment]
@AndySDH can you please open a fresh issue so it can be tracked

**Pull request merged**